### PR TITLE
Support Image Capture feature

### PR DESCRIPTION
**Pull request by lbegani**
Image Capture will be triggered by mavlink messages.

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -259,7 +259,10 @@ BASE_FILES += \
 	src/CameraServer.h \
 	src/CameraDevice.h \
 	src/CameraDevice_V4L2.cpp \
-	src/CameraDevice_V4L2.h
+	src/CameraDevice_V4L2.h \
```
Comment (shakthi-prashanth-m) [inline Makefile.am]: Filename uses mix of CamelCase & snake_case. Please change it CamelCase as it is consistent with existing source files. Same applies to class names as well.

File: src/CameraComponent.h [hunk 2]
```diff
@@ -68,6 +70,10 @@ class CameraComponent {
                          size_t value_size, int param_type);
     virtual int setCameraMode(uint32_t mode);
     virtual int getCameraMode();
+    virtual int startImageCapture(int interval, int count,
+                                  std::function<void(int result, int seq_num)> cb);
```
Comment (shakthi-prashanth-m) [inline src/CameraComponent.h]: You can define callback type like this, 
`typedef std::function<void(int result, int seq_num)> image_capture_callback_t;`

File: src/CameraDevice.h [hunk 3]
```diff
@@ -42,6 +42,7 @@ class CameraDevice {
         STATE_ACTIVE = 1,
     };
 
+    virtual std::string getDevice() = 0;
```
Comment (shakthi-prashanth-m) [inline src/CameraDevice.h]: Method `CameraComponent::getDevice()` doesn't intuitively tell whether it returns device name.
I prefer to call `CameraComponent::getDeviceName()`

File: src/ImageCapture_Gst.h [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    int create_v4l2_pipeline(int seq_num);
+    std::string getGstElementName_format(int format);
+    std::string getExtension(int format);
+    std::string getGstPipelineName_v4l2(int seq_num);
+    std::shared_ptr<CameraDevice> mCamDev;
+    std::string mDevice;
+    std::atomic<int> mState;
+    int mWidth;
+    int mHeight;
+    int mImgFormat;
```
Comment (shakthi-prashanth-m) [inline src/ImageCapture_Gst.h]: Its better to use `enum` for image format & pixel format over `int`, which increases readability.

File: src/ImageCapture_Gst.cpp [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+
+int ImageCapture_Gst::setDestination(const std::string imgPath)
+{
+    // TODO::Check if the path is writeable
+    mImgPath = imgPath;
+
+    return 0;
+}
+
+void ImageCapture_Gst::capture_thread(int num, int interval)
```
Comment (shakthi-prashanth-m) [inline src/ImageCapture_Gst.cpp]: Method is in snake_case but others are in CamelCase; I suggest `ImageCaptureGst::start_capture_thread()`.

File: Makefile.am (hunk 1 above)
Comment (lbegani) [reply Makefile.am]: Fixed

File: src/CameraComponent.h (hunk 2 above)
Comment (lbegani) [reply src/CameraComponent.h]: Fixed

File: src/CameraDevice.h (hunk 3 above)
Comment (lbegani) [reply src/CameraDevice.h]: Fixed, its not name but Id

File: src/ImageCapture_Gst.cpp (hunk 5 above)
Comment (lbegani) [reply src/ImageCapture_Gst.cpp]: Fixed

File: src/ImageCapture_Gst.h (hunk 4 above)
Comment (lbegani) [reply src/ImageCapture_Gst.h]: Possible values for the formats are defined as const integers, hence int

**Pull request merged**